testclient.cpp

```diff
@@ -23,6 +26,10 @@ static int printUsage() {
 
 int main( int argc, char** argv ) {
     QCoreApplication app( argc, argv );
+#ifdef __HAIKU__
+    // XXX: for some reason the QApp object doesn't create a BApp
+    BApplication bapp("application/x-vnd.QtKeychain");
```

Comment:
How do other Qt application do it?  Maybe it only works with QApplication or QGuiApplication?
If that's the case, then better change to QApplication for haiku.

Reply:
I'll try with it.
Either that or it is just created later on if we actually start it.
In Haiku as well sometimes we create the app object without calling Run() on it because it's not required for what we need.